tests/client/core/test_utils.js

```diff
@@ -172,56 +180,150 @@ describe('getClientApp', () => {
-    assert.strictEqual(clientSupportsAddons(ua), true, `UA string: ${ua}`);
-  });
-
   it('returns true for Firefox OS', () => {
-    userAgents.firefoxOS.forEach((ua) => {
-      assert.strictEqual(clientSupportsAddons(ua), true, `UA string: ${ua}`);
+    userAgents.firefoxOS.forEach((userAgent) => {
+      assert.deepEqual(
+        isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
+        { compatible: true },
+        `UA string: ${userAgent}`);
     });
   });
 
   it('returns false for Firefox iOS', () => {
-    userAgents.firefoxIOS.forEach((ua) => {
-      assert.strictEqual(clientSupportsAddons(ua), false, `UA string: ${ua}`);
+    userAgents.firefoxIOS.forEach((userAgent) => {
+      assert.deepEqual(
+        isCompatibleWithUserAgent({ userAgentInfo: UAParser(userAgent) }),
+        { compatible: false, reason: INCOMPATIBLE_FIREFOX_FOR_IOS },
+        `UA string: ${userAgent}`);
     });
   });
 
+  it('should mark non-Firefox UAs as incompatible', () => {
+    const userAgentInfo = { browser: { name: 'Chrome' } };
+    assert.deepEqual(
+      isCompatibleWithUserAgent({ userAgentInfo }),
+      { compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX });
+  });
+
+  it('should mark Firefox 10 as incompatible with a minVersion of 10.1', () => {
+    const userAgentInfo = {
+      browser: { name: 'Firefox', version: '10.0' },
+      os: { name: 'Windows' },
+    };
+    assert.deepEqual(isCompatibleWithUserAgent({
+      minVersion: '10.1', userAgentInfo }),
+      { compatible: false, reason: INCOMPATIBLE_UNDER_MIN_VERSION });
+  });
+
+  it('should mark Firefox 24 as incompatible with a maxVersion of 8', () => {
+    const userAgentInfo = {
+      browser: { name: 'Firefox', version: '24.0' },
+      os: { name: 'Windows' },
+    };
+    assert.deepEqual(isCompatibleWithUserAgent({
+      maxVersion: '8', userAgentInfo }),
+      { compatible: false, reason: INCOMPATIBLE_OVER_MAX_VERSION });
+  });
+
+  it('should mark Firefox as compatible when no min or max version', () => {
+    const userAgentInfo = {
+      browser: { name: 'Firefox', version: '10.0' },
+      os: { name: 'Windows' },
+    };
+    assert.deepEqual(isCompatibleWithUserAgent({ userAgentInfo }),
+      { compatible: true });
+  });
+
+  it('should mark Firefox as compatible with maxVersion of "*"', () => {
+    // WebExtensions are marked as having a maxVersion of "*" by addons-server
+    // if their manifests don't contain explicit version information.
+    const userAgentInfo = {
+      browser: { name: 'Firefox', version: '54.0' },
+      os: { name: 'Windows' },
+    };
+    assert.deepEqual(isCompatibleWithUserAgent({
+      maxVersion: '*', userAgentInfo }),
+      { compatible: true });
+  });
+
+  it('should log warning when minVersion is "*"', () => {
+    // Note that this should never happen as addons-server will mark a
+    // WebExtension with no minVersion as having a minVersion of "48".
```

Comment:
It might still happen for non-webextension add-ons or for webextensions that specify `*` explicitly in their manifest but I don't know